src/index.js

```diff
@@ -149,6 +152,37 @@ module.exports = function(source, inputSourceMap) {
     options.sourceFileName = relative(options.sourceRoot, options.filename);
   }
 
+  if (this.version > 1 && options.presets) {
+    options.presets.forEach(preset => {
+      let name = preset;
+      let opts = {};
+
+      // if we have extra options put it in opts
+      if (Array.isArray(preset)) {
+        name = preset[0];
+        opts = preset[1];
+      }
+
+      if (
+        ["es2015", "env", "latest"].indexOf(name) > -1 &&
+        (!emitCache.has(options.presets) &&
+          (!opts ||
+            (opts && !opts.hasOwnProperty("modules")) ||
```

Comment:
I'd avoid `opts.hasOwnProperty` as an instance property. Also you never validate that `opts` is an object, so this would throw if someone passed an invalid config instead of passing through to `babel-core`'s validation. I'd go with

```
const PRESET_WHITELIST = new Set(["es2015", "env", "latest"]);

// ...

const { modules } = typeof opts === "object" && opts || {};

if (PRESET_WHITELIST.has(name) && modules !== false) {
  this.emitWarning // ...
}
```